src/federation.rs

```diff
@@ -192,6 +210,30 @@ impl Federation {
 
         Ok(())
     }
+
+    pub fn from(pubkey: PublicKey, ser: SerFederation) -> Self {
+        Self::new(pubkey, ser.block_height, ser.threshold, ser.nodevss)
+    }
+
+    pub fn to_ser(self) -> SerFederation {
+        SerFederation {
+            block_height: self.block_height,
+            threshold: self.threshold,
+            nodevss: self.nodevss,
+        }
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct SerFederations {
```

Comment:
I think SerFederations or SerFederation is not needed if we use Federations or Federation instead and change the federation file format.

Reply:
I think it is appropriate idea for current implementation but I'm wondering that the Federation struct would have some new fields for caching calculated value like an aggregated public key.

We have some options for that. For example add the cache fields to Federation and set attribute for skipping serialization on serde functionality. 
However I thought the way this PR is using was most simple for me.